### PR TITLE
Create PULL_REQUEST_TEMPLATE.md

### LINKED ISSUE #1067: Define and Setup PR template

### PR DESCRIPTION
**Pull request by agnivade**
Closes #1067

### CODE REVIEW

**Comment by agnivade** [comment]
@waldyrious / @ostera / @sbrl -  These are all the points I could gather. Please let me know if you guys think there is anything else to add.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+Thank you for sending a PR !
+
+Please perform the following checks and tick mark all the points.
+
+- [ ] If its a new page, it does not already exist in the repo.
+
+- [ ] If its a new page, it is added to the correct platform folder.
+i.e. If its common to all platforms, it is under `common/` and if its linux specific, its under `linux/`. If the tool works on all unix-y platforms, please add multiple copies of the same page under all folders.
+
+- [ ] The total no. of examples in the page are under 8.
```
Comment (sbrl) [inline .github/PULL_REQUEST_TEMPLATE.md]: I feel that "no." should be written out in full.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Thank you for sending a PR !
+
+Please perform the following checks and tick mark all the points.
+
+- [ ] If its a new page, it does not already exist in the repo.
```
Comment (sbrl) [inline .github/PULL_REQUEST_TEMPLATE.md]: "its" should have an apostrophe here. See http://www.bristol.ac.uk/arts/exercises/grammar/grammar_tutorial/page_13.htm

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+
+- [ ] If its a new page, it is added to the correct platform folder.
```
Comment (sbrl) [inline .github/PULL_REQUEST_TEMPLATE.md]: See about above about `its`.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -0,0 +1,10 @@
+Thank you for sending a PR !
```
Comment (ostera) [inline .github/PULL_REQUEST_TEMPLATE.md]: This is nice :) but could be included as a comment so it's invisible when publishing the PR. As in

```html
<!-- comments! -->
```

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (agnivade) [inline .github/PULL_REQUEST_TEMPLATE.md]: Ah, I always mix up the two. Thanks !

**Comment by agnivade** [comment]
> I'd say the overall tone should be more statements than instructions, like in the last checkbox instead of the first ones.

Hmm .. its the part about the page being a new page that I am finding a bit hard to convert to a statement. Does this work @ostera ? 

```
The page(if new), is added to the correct platform folder
```

**Comment by ostera** [comment]
@agnivade wouldn't those statements apply regardless of the page being new or not?

**Comment by agnivade** [comment]
@ostera -  If you are just editing a page, then you are not changing its location. So, the question of checking its location does not arise.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (sbrl) [inline .github/PULL_REQUEST_TEMPLATE.md]: Np :smiley_cat:

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -0,0 +1,10 @@
+<!-- Thank you for sending a PR !-->
+
+Please perform the following checks and tick mark all the points.
+
+- [ ] The page (if new), does not already exist in the repo.
+
+- [ ] The page (if new), is added to the correct platform folder.
+i.e. If its common to all platforms, it is under `common/` and if its linux specific, its under `linux/`. If the tool works on all unix-y platforms, please add multiple copies of the same page under all folders.
```
Comment (sbrl) [inline .github/PULL_REQUEST_TEMPLATE.md]: These `its` need an apostrophe.

 - "If its common to all platforms" -> "If it's common to all platforms"
 - "and if its linux specific, its under" -> "and if it's linux specific, it's under"

**Comment by agnivade** [comment]
ping @ostera. Do the latest changes look good ?

**Comment by waldyrious** [comment]
Sorry for the delay. I'll leave minor wording comments inline, but my main comment, from my experience with projects with similar automated checklists, is that it's important to mention how to proceed when the checks don't apply to the PR -- for example, PRs editing package.json, or fixing a typo in the README, etc.

I'd suggest adding instructions to leave those unchecked, and apply strike-through formatting, as I did in https://github.com/monicahq/monica/pull/522. Thoughts?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+<!-- Thank you for sending a PR !-->
```
Comment (waldyrious) [inline .github/PULL_REQUEST_TEMPLATE.md]: I've seen @ostera's [comment](https://github.com/tldr-pages/tldr/pull/1434#discussion_r128926550), but I don't understand why this line is marked as a comment and the next one isn't. I'd understand if this line was directed at the PR author and the others not (e.g. if the line "Please perform the following checks..." was something similar to "This submission conforms to the project's requirements for PRs", which could be read by both the PR author and other people), but with the current wording, it isn't clear why there's this distinction.

Perhaps this aligns with what @ostera meant about the tone of the message.

My suggestion would be to either comment out both lines and leave only the checks uncommented, or leave the first line commented out and change the wording of the second one to a wording similar to what I suggested above.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+
+Please perform the following checks and tick mark all the points.
```
Comment (waldyrious) [inline .github/PULL_REQUEST_TEMPLATE.md]: Suggestion: "tick mark all the points." --> "check all the boxes that apply." (This sounds more idiomatic to me, but I'm open to other opinions.)

By the way, here would be a good place to include the instructions for when the boxes don't apply :)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+
+- [ ] The page (if new), does not already exist in the repo.
+
+- [ ] The page (if new), is added to the correct platform folder.
```
Comment (waldyrious) [inline .github/PULL_REQUEST_TEMPLATE.md]: "is added" --> "has been added".

Also include "(linux, osx, etc.)" at the end, just to be sure we're not using jargon the submitter may be unsure about.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 6 above)
Comment (agnivade) [inline .github/PULL_REQUEST_TEMPLATE.md]: Agree. Both lines should be commented.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 9]
```diff
@@ -0,0 +1,10 @@
...
+i.e. If it's common to all platforms, it is under `common/` and if it's linux specific, it is under `linux/`. If the tool works on all unix-y platforms, please add multiple copies of the same page under all folders.
```
Comment (waldyrious) [inline .github/PULL_REQUEST_TEMPLATE.md]: This renders correctly as-is, but for extra neatness, I'd suggest indenting the line to align with the text of the line above, and wrapping it by inserting a line break after "\`linux/\`."

Here too I'd suggest changing "it is under" to some other wording, perhaps something simpler to piggyback on the first sentence:

    - [ ] The page (if new), is added to the correct platform folder,
          i.e. `common/` common to all platforms, `linux/` if it's Linux-specific, and so on.

I think the line about working the same on unix-y platforms is too specific -- we could leave such pages on `common/`, which IIRC is already the case for the majority of pages in that folder.

At most, we could add a line about tools that do exist in multiple platforms but with subtle differences (tipically linux vs mac), but that too seems overly specific and wordy for a generic PR template, IMO.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 10]
```diff
@@ -0,0 +1,10 @@
...
+
+- [ ] The total number of examples in the page are under 8.
```
Comment (waldyrious) [inline .github/PULL_REQUEST_TEMPLATE.md]: Slight rewording suggestion: "The page has 8 or fewer examples"

**Comment by waldyrious** [comment]
Re-reading [my notes from #1067](https://github.com/tldr-pages/tldr/issues/1067#issuecomment-248833074), I think most of those bullet points should actually be in the contributing guidelines, but two of them should be added here:

1. That the page follows the guidelines (maybe we could split the ["markdown format" section](https://github.com/tldr-pages/tldr/blob/master/CONTRIBUTING.md#markdown-format) to a separate file within, say, a `docs/` folder, so it could be linked directly from CONTRIBUTING.md and from the PR template. What do you guys think?)
2. The note about how to title the PR ("command: add page" for new pages, or "command: description of changes" for pages being edited)

**Comment by agnivade** [comment]
@waldyrious - Thank you for the review ! PTAL.

**Comment by waldyrious** [comment]
@agnivade apologies for the delay. I've made a few adjustments, PTAL. If it looks good to you guys, the PR can be merged as far as I'm concerned. Nice work! :+1:

**Comment by agnivade** [comment]
Looks great, let's go !

**Pull request merged**